build.gradle

```diff
@@ -18,82 +17,81 @@ allprojects {
     }
 }
 
-def problemVersion = '0.23.0'
-def jerseyVersion = '2.22.2'
-
-dependencies {
-    implementation group: 'com.github.BIBSYSDEV', name: 'nva-commons', version: 'v0.3.2'
-    implementation group: 'com.amazonaws', name: 'aws-lambda-java-core', version: '1.2.0'
-    implementation group: 'com.amazonaws', name: 'aws-java-sdk-s3', version: '1.11.715'
-    implementation group: 'org.glassfish.jersey.core', name: 'jersey-common', version: jerseyVersion
-    implementation group: 'org.zalando', name: 'problem', version: problemVersion
-    implementation group: 'org.zalando', name: 'jackson-datatype-problem', version: problemVersion
-
-    testImplementation group: 'com.github.BIBSYSDEV', name: 'nva-testutils', version: 'v0.1.6'
-    testImplementation group: 'junit', name: 'junit', version: '4.12'
-    testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.2.4'
-    testImplementation group: 'commons-io', name: 'commons-io', version: '2.6'
-}
+subprojects {
+    apply plugin: 'java-library'
+    apply plugin: 'jacoco'
+    apply plugin: 'checkstyle'
+    apply plugin: 'pmd'
 
-test {
-    failFast = true
-    finalizedBy jacocoTestReport
-}
+    sourceCompatibility = JavaVersion.VERSION_11  // source-code version and must be <= targetCompatibility
+    targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
 
-task buildZip(type: Zip) {
-    from compileJava
-    from processResources
-    into('lib') {
-        from configurations.runtimeClasspath
+    dependencies {
+        testImplementation group: 'com.github.BIBSYSDEV', name: 'nva-testutils', version: 'v0.1.7'
+        testImplementation group: 'junit', name: 'junit', version: '4.12'
+        testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.2.4'
     }
-}
 
-project.tasks.build.dependsOn project.tasks.buildZip
+    test {
+        failFast = true
+        finalizedBy jacocoTestReport
+    }
 
-pmd {
-    ruleSetConfig = rootProject.resources.text.fromFile('config/pmd/ruleset.xml')
-    ruleSets = []
-    ignoreFailures = false
-}
+    task buildZip(type: Zip) {
```

Comment:
is this a fat-jar?

Reply:
No, and I don't know what this is used for. I will remove it.